samples/default-hierarchy-gallery-mobile/local-check.sh

```diff
@@ -0,0 +1,34 @@
+#
+# Copyright 2024 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license.
+#
+
+set -e
+
+log() {
+  echo "\033[0;32m> $1\033[0m"
+}
+
+./gradlew clean assembleDebug
+log "default-hierarchy-gallery-mobile android success"
+
+if ! command -v xcodebuild &> /dev/null
+then
+    echo "xcodebuild could not be found, skip ios checks"
+    log "default-hierarchy-gallery-mobile checked"
+
+    exit 0
+fi
+
+./gradlew clean compileKotlinIosX64
+log "default-hierarchy-gallery-mobile ios success"
+
+./gradlew clean podspec build generateDummyFramework --rerun-tasks
```

Comment:
we skip tests run in this case on nonmacos.
lets add 
```
else
    ./gradlew build
    log "default-hierarchy-gallery-mobile full build success"
```
and in other script do same logic. in case when we not on macos - we still should run all available build and tests.

Reply:
rework scripts